src/shell/mod.rs

```diff
@@ -433,14 +433,13 @@ macro_rules! static_command {
             // thus static, and the_function is static by construction as well)
             unsafe impl Sync for StaticCommand {}
 
+            #[link_section = ".roxfa.shell_commands_xfa.5"]
+            #[export_name = concat!("shell_commands_xfa_5_", stringify!($modname))]
```

Comment:
Doing this backward compatible would probably require some cooperation from RIOT. I guess changing the XFA section name (e.g. from `.roxfa.shell_commands_xfa.5` to `.roxfa.shell_commands_v2_xfa.5`) and providing both ptr and element would work, assuming that the linker would garbage collect the ptr in the newer version.

Reply:
That seems to indeed just work

Reply:
The garbage collection does not work here, so while this does allow adding shell commands in a way that works with both pre and post !20958 RIOT, it will not reduce the overhead.

I guess this is acceptable mid-term? We could drop the ptr for backward compatibility in the long run.

Reply:
Dropping this on the long run sounds fine. This is shell commands, and a shell is not what you do when you're starved for every single pointer in RAM.